File: 3360-minimum-deletions-to-make-string-k-special/minimum-deletions-to-make-string-k-special.js
/**
 * @param {string} word
 * @param {number} k
 * @return {number}
 */
var minimumDeletions = function(word, k) {
        const map= new Map();
        for(const ch of word){
            map.set(ch,(map.get(ch)||0)+1);
        }

        let ans= word.length;

        for(const a of map.values()){
            let deleted=0;
            for (const b of map.values()){
                if(a>b){
                    deleted+=b;
                }
                else if (a+k<b){
                    deleted+=b-a-k;
                }
            }
            ans =Math.min(deleted,ans);
        }
        return ans;
};

File: 3360-minimum-deletions-to-make-string-k-special/minimum-deletions-to-make-string-k-special.cpp
class Solution {
public:
    int minimumDeletions(string word, int k) {
        unordered_map<char,int> mp;
        for(char x : word){
            mp[x]++;
        }
        int res= word.size();
        for( auto &[_,a]:mp){
            int deleted = 0;
             for( auto &[_,b]:mp){
                if(b<a){
                    deleted+=b;
                }
               else if (b > a + k) {
                    deleted += b - (a + k);
                }
             }
             res=min(res,deleted);
        }
        return res;
    }
};

